### ISSUE TITLE
Improve auto update feature

### ISSUE THREAD

**Issue by JeroenVinke** [monterey-framework/monterey#328]
We have discussed this before, just formalizing it in an issue

### Intention

**I'm submitting a feature request**  

### Description
**Current behavior:**  
Monterey automatically updates and does not show changelog. You do not have the ability to decline an update

**Expected/desired behavior:**  
Monterey should present the user with a notification that an update is available. Should we have an notification center button in the taskbar so we can show other type of notifications as well? Users should be able to read the changelog and then decide whether or not to update.

**Comment by Thanood** [comment]
@JeroenVinke Which version does actually auto-update?
I have v0.3.1 installed.

**Comment by JeroenVinke** [comment]
monterey logs all messages of the auto-updater to console, so it should tell you what is happening

**Comment by Thanood** [comment]
It's updated to v.0.3.2!  \o/

Complaining about not being able to read the change log from the NuGet file. But I guess that's the point of this issue.
Also, there was no notification.

**Comment by JeroenVinke** [comment]
probably because the toastr message disappears too quickly. We need a way to show notifications that user specifically can mark as "read" and does not disappear after just a few seconds 😄

**Comment by adriatic** [comment]
Assuming that you use toastr.js widget and further assuming that it is a created by someone with the proper sense for logic, this

`toastr.options.closeButton = true;`

should do what is needed 😏

**Comment by Thanood** [comment]
From an older app:

```javascript
let promise = new Promise(resolve => {
  let toast = toastr.info('Monterey updated to v0.3.2 - click to dismiss', 'Update applied', {
    timeOut: 10000,
    onClick: resolve(toast)
  });
});
promise.then((toast) => {
  toast.remove();
});
```

**Comment by Thanood** [comment]
Uhm.. Or what @adriatic says. :joy:

**Comment by JeroenVinke** [comment]
Yes that would solve the issue, but it may introduce a different issue where too many notifications decide to stay on screen, and the user ends up having to close these messages because they are taking up too much space of their screen.

**Comment by JeroenVinke** [comment]
I am thinking of being able to display other notifications as well, for example when there is a new post on the Aurelia blog

**Comment by adriatic** [comment]
I am surprise by your comment, Jeroen as I did not suggest to use `toastr.options.closeButton = true;` indiscriminantly. Use it only in cases where user's acknowledgment is meaningful

**Comment by adriatic** [comment]
> I am thinking of being able to display other notifications as well, for example when there is a new post on the Aurelia blog

Excellent iidea!!! I would recommend that now, having the first Monterey done, we should all wear our marketing hats all the time, as you all know that it might be more important that any other thing we do around monterey.

**Comment by JeroenVinke** [comment]
> I am surprise by your comment, Jeroen as I did not suggest to use toastr.options.closeButton = true; indiscriminantly. Use it only in cases where user's acknowledgment is meaningful

I got that message, but my point is that if we have too many of these cases then it becomes a problem. Seeing that you liked my idea about a notification center,  I will open up a different issue for it

**Comment by adriatic** [comment]
> I got that message, but my point is that if we have too many of these cases then it becomes a problem. 

Shame on me for even doubt that 😢

**Comment by adriatic** [comment]
> Seeing that you liked my idea about a notification center, I will open up a different issue for it

##I love it

**Comment by JeroenVinke** [comment]
we should also create a good test setup: https://github.com/monterey-framework/monterey/issues/222

**Comment by adriatic** [comment]
A summary on notification center in the context of auto-updater messages:

1. Monterey should maintain the auto-updater specific log which is always updated at the top (not appended at the bottom). 

2. In addition to that auto-updater uses the notification center to show the very brief summary of the update in the auto-updater specific color (this might require to switch from current notifications to `http://alertifyjs.com/`)

3. The `About` button on toolbar should privide the UI for reading the updater log (having the ability to search would be nice)
